From 97a0b28b83af7554dbce5f9357f10a1f4aa04e78 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:58:57 -0400
Subject: [net] core: Add sysfs file for port number

Message-id: <39bf39919b6887d0137d29207e9a49c502cdbb10.1410425016.git.avadai@redhat.com>
Patchwork-id: 93627
O-Subject: [Patch RHEL7.1 V1 001/197] net: Add sysfs file for port number
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Amir Vadai <amirv@mellanox.com>

Add a sysfs file to enable user space to query the device
port number used by a netdevice instance. This is needed for
devices that have multiple ports on the same PCI function.

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 3f85944fe207d0225ef21a2c0951d4946fc9a95d)

The new member "dev_port" is placed in a free space (after the u16
dev_id), therefore kABI didn't break.

CC: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/netdevice.h b/include/linux/netdevice.h
index 1fafa0a..4eab278 100644
--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -1239,7 +1239,12 @@ struct net_device {
  unsigned char  addr_len; /* hardware address length */
  unsigned short  neigh_priv_len;
  unsigned short          dev_id;  /* for shared network cards */
-
+#ifndef __GENKSYMS__
+ unsigned short          dev_port; /* Used to differentiate
+       * devices that share the same
+       * function
+       */
+#endif
  spinlock_t  addr_list_lock;
  struct netdev_hw_addr_list uc; /* Unicast mac addresses */
  struct netdev_hw_addr_list mc; /* Multicast mac addresses */
diff --git a/net/core/net-sysfs.c b/net/core/net-sysfs.c
index 3f40ea9..62086f2 100644
--- a/net/core/net-sysfs.c
+++ b/net/core/net-sysfs.c
@@ -104,6 +104,7 @@ static ssize_t netdev_store(struct device *dev, struct device_attribute *attr,
 }
 
 NETDEVICE_SHOW_RO(dev_id, fmt_hex);
+NETDEVICE_SHOW_RO(dev_port, fmt_dec);
 NETDEVICE_SHOW_RO(addr_assign_type, fmt_dec);
 NETDEVICE_SHOW_RO(addr_len, fmt_dec);
 NETDEVICE_SHOW_RO(iflink, fmt_dec);
@@ -373,6 +374,7 @@ static struct attribute *net_class_attrs[] = {
  &dev_attr_netdev_group.attr,
  &dev_attr_type.attr,
  &dev_attr_dev_id.attr,
+ &dev_attr_dev_port.attr,
  &dev_attr_iflink.attr,
  &dev_attr_ifindex.attr,
  &dev_attr_addr_assign_type.attr,
-- 
1.7.1